Source/WebCore/platform/network/BlobRegistryImpl.cpp

```diff
@@ -120,6 +120,34 @@ void BlobRegistryImpl::registerFileBlobURL(const URL& url, Ref<BlobDataFileRefer
     addBlobData(url.string(), WTFMove(blobData));
 }
 
+Ref<DataSegment> BlobRegistryImpl::createDataSegment(Vector<uint8_t>&& movedData, BlobData& blobData)
+{
+    ASSERT(isMainThread());
+
+    auto data = DataSegment::create(WTFMove(movedData));
+    if (m_fileDirectory.isEmpty())
+        return data;
+
+    static uint64_t blobMappingFileCounter;
+    static NeverDestroyed<Ref<WorkQueue>> workQueue(WorkQueue::create("BlobRegistryImpl Data Queue"));
+    auto filePath = FileSystem::pathByAppendingComponent(m_fileDirectory, makeString("mapping-file-", ++blobMappingFileCounter, ".blob"));
+    workQueue.get()->dispatch([blobData = Ref { blobData }, data, filePath = WTFMove(filePath).isolatedCopy()]() mutable {
+        auto mappedFileData = FileSystem::mapToFile(filePath, data->size(), [&data](auto& writeData) {
+            writeData({ data->data(), data->size() });
+        });
+        if (!mappedFileData)
+            return;
+
+        FileSystem::deleteFile(filePath);
```

Comment:
If you delete the file before you write the bytes, can you enable this technique in ephemeral sessions too?

Reply:
I tried it in the latest patch and this indeed works